ethpm/tools/builder.py

```diff
@@ -341,6 +342,25 @@ def _contract_type(
     return assoc_in(manifest, ["contract_types", name], contract_type_data)
 
 
+@to_dict
+def filter_all_data_by_selected_fields(
+    all_type_data: Dict[str, Any], selected_fields: List[str]
+) -> Iterable[Tuple[str, Any]]:
+    """
+    Raises exception if selected field data is not available in the contract type data
+    automatically gathered by normalize_compiler_output. Otherwise, returns the data.
+    """
+    for field in selected_fields:
+        if field in all_type_data:
+            yield field, all_type_data[field]
+        else:
+            raise ManifestBuildingError(
+                f"Selected field: {field} not available in data collected from solc output: "
+                f"{list(all_type_data.keys())}. Please make sure the relevant data is present "
```

Comment:
*nitpick*

since dictionary keys aren't ordered this can produce two different messages with the same data.  Normally when I dump unordered data into a logging/user message I first sort it.